src/System.Private.Interop/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -1029,8 +1024,27 @@ private static unsafe void PtrToStructureHelper(IntPtr ptr, Object structure)
 
             // Boxed struct start at offset 1 (EEType* at offset 0) while class start at offset 0
             int offset = structureTypeHandle.IsValueType() ? 1 : 0;
+            bool useMemCpy = false;
+
+            if (RuntimeInteropData.Instance.TryGetStructUnmarshalStub(structureTypeHandle, out IntPtr unmarshalStub))
```

Comment:
This check has to be first to handle non-blitable types without GC references correctly. This is bug in ProjectN too.